specs/Deployment/MSIXPackages.md

```diff
@@ -0,0 +1,189 @@
+[startup tasks](https://docs.microsoft.com/uwp/api/Windows.ApplicationModel.StartupTask) executing
+on user login, [ApplicationData](https://docs.microsoft.com/uwp/api/Windows.Storage.ApplicationData)
+for data storage scoped to Windows App SDK (not any particular app using Windows App SDK), and
+more.
+
+The primary scenarios include:
+
+* Brokered Access -- processes running in an [AppContainer](https://docs.microsoft.com/windows/win32/secauthz/appcontainer-isolation) requiring
+                     access to resources that cannot be access from within the AppContainer can use
+                     a 'helper process' running at MediumIL and doing work on the API's behalf.
+                     cannot do for itself (i.e. brokered access to a resource).
+* Long-Running Process (aka user service) -- functionality requiring a long-running process
+                                             (especially if its lifetime isn't directly coupled to
+                                             an app's lifetime).
+
+Main packages don't support side-by-side behavior for a user. A user can only have 1 (or 0) Main
+packages in a package family registered at a time. Thus if a user has v1.0.0.0 registered when
+1.0.0.1 is registered it's an *update*, ending with the user having v1.0.0.1 registered instead of
+v1.0.0.0. This poses servicing challenges, as updating the Main package requires any running
+processes from the existing Main package to be terminated.
+
+For these reasons developers are strongly encouraged to only rely on the Main package when
+necessary.
+
+If a process from the Main package is necessary, it should be short-lived or support a save/restore
+execution model. The former shrinks the possibility a running process when a servicing event
+occurs. The latter ensures if Main package processes are running when a servicing event occurs, the
+running processes can be terminated (e.g. via [DeploymentOptions.ForceTargetApplicationShutdown](https://docs.microsoft.com/uwp/api/windows.management.deployment.deploymentoptions))
+to allow the servicing event to occur and restarted upon its completion, with only a transient (and
+possibly unnoticed) interruption of service.
+
+Further, if/when the Main package is necessary, developers are encouraged to put as
+little as practical in the Main package and use the Framework package as much as possible. For
+example, if a StartupTask is needed you can declare the StartupTask in the Main package's
+`appxmanifest.xml` and provide an executable with a trivial 1% implementation e.g.
+
+```c
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, PWSTR pCmdLine, int nCmdShow)
+{
+    return MyFeatureWinMain(hInstance, hPrevInstance, pCmdLine, nCmdShow);
+}
+```
+
+and export `MyFeatureWinMain` from a DLL in the Framework package to get the implementation's
+other 99%.
+
+### Role - Singleton
```

Comment:
is this per user?

Reply:
Yes. All MSIX packages are registered per-user.

The installer will also provision all the packages if it can (i.e. if it's running with admin privilege i.e. elevated), but provisioning just makes packages available to all users. That doesn't change the core principle of MSIX as a per-user app model.

We have feature today needing machine-wide singleton semantics. We may in the future. We also have more tools at our disposal we haven't needed to leverage (so far) e.g. Packaged Services. We've considered the issue but lacking a driving need we haven't seen the need to drive any sort of stake in the ground how we'll implement said unknown theoretical future features :-)